Day 15: “The Wonderful World of Web Feeds” / Maureen Holland

maureenlholland opened the pull request:
Draft for Web Feeds and Where to Find Them ready for review: http://localhost:8080/adventcalendar/2025/15/

maureenlholland committed:
Add advent15 post

ZoeBijl commented:
thank you for your contribution, before i begin the review, would you consider changing the title to something not related to J.K. Rowling?

maureenlholland commented:
@ZoeBijl sure! I hadn't realized it was making that connection. I just liked the alliteration and it helped me structure the article. 

Now all that's in my head are parody titles 🫠 
Where the Web Feeds Are 
The Wonderful World of Web Feeds

Happy to take suggestions here and will keep thinking about it

maureenlholland committed:
Update title and remove now-irrelevant headings

maureenlholland commented:
Updated! (Although very tempted by "A Webful of Feeds" 🤠 🕸️ )

matuzo reviewed (commented):
Hey Maureen,

thank you for your post. I read this as someone who has no idea what feeds are. I (still) have the following questions:

1. What is a feed and why do I need it? You say "It is an XML-based specification for web content syndication", but what is web content syndication and what do I need it for?
2. What's a feed reader?
3. Is RSS Autodiscovery something that I need to implement or install?
4. _Can_ I include separate links for different formats or do I _have to_? Like, would having a rss.xml suffice?

maureenlholland committed:
Draft 2

- Change RSS "oldest and most common" to "oldest" (I thought I read
it was most common somewhere but can't find the source now and it
might be out of date information at this point anyway)
- More explanation upfront about web content syndication and feed
reader applications
- Remove "two main reasons" section and incorporate the content
elsewhere (I just kept bumping on it as I tried to read the whole
article, it felt out of place)
- Describe some of the XML example in prose
- Remove multiple formats example (on the fence about this, but
thought it was a bit distracting and, although I've seen sites do
this, I don't entirely understand why you would include multiple
feed formats)
- Expand categories example

maureenlholland commented:
Second draft ready for review (I made more changes than I anticipated, let me know if it's gone too far and I should iterate more closely on the first draft)

matuzo commented:
> Second draft ready for review (I made more changes than I anticipated, let me know if it's gone too far and I should iterate more closely on the first draft)

Fantastic, thank you!

ZoeBijl commented on hell/adventcalendar/2025/15/index.md:
```diff
@@ -1,35 +1,99 @@
-Some text.
 
-Some text. Some text.
+Web feeds are incredible! And a bit confusing! Why are the feed links often called “RSS”? And why is this “RSS” feed in an <code>atom.xml</code> file… hang on, what is <code>feed.json</code> for?  What are they even feeding into anyway?
```
remove double space.

```suggestion
Web feeds are incredible! And a bit confusing! Why are the feed links often called “RSS”? And why is this “RSS” feed in an <code>atom.xml</code> file… hang on, what is <code>feed.json</code> for? What are they even feeding into anyway?
```

ZoeBijl commented on hell/adventcalendar/2025/15/index.md:
```diff
@@ -1,35 +1,99 @@
+To start, web feeds are often referred to as “RSS” because RSS is the oldest format. [RSS](https://www.rssboard.org/rss-specification) stands for Really Simple Syndication. It is an XML-based specification for web content syndication ([including podcasts](https://help.apple.com/itc/podcasts_connect/#/itcb54353390)).
 
-Some text.
+In [broadcast syndication](https://web.archive.org/web/20091009143514/http://www.museum.tv/archives/etv/S/htmlS/syndication/syndication.htm), networks sell reruns of their original shows to other platforms. In web content syndication, feeds package web content into a format that can “rerun” on a feed reader application. Notably, web content creators are not selling their content to feed readers.  ...
```
As a non-native speaker I had to look up “syndication”. Which might be fine but it could be nice to explain it inline. I really like the comparison with broadcast syndication and think that could be expanded upon for this example.

Something along the lines of:

```suggestion
Syndication is the sale or licensing of material for publication or broadcasting by others. This system generall has three parts: a seller, a buyer, and a consumer. In [broadcast syndication](https://web.archive.org/web/20091009143514/http://www.museum.tv/archives/etv/S/htmlS/syndication/syndication.htm), networks sell reruns of their original shows to other platforms, the buyers. The consumer then watches the rerun on the buyer’s channel. In web content syndication feeds are the sellers. They package web content into a format that can be “rerun” by the buyer using a feed reader application. Notably, web content creators are not selling their content to feed readers. Most feeds and readers are free. With web feeds you’re both the buyer and the consumer. You choose what content you want to follow and what reader you want to follow it on.
```
This is just a suggestions of course :). Feel free to ignore or rewrite in your own style.

ZoeBijl commented on hell/adventcalendar/2025/15/index.md:
```diff
@@ -1,35 +1,99 @@
+
+It uses the unique identifier to determine if an item in the feed is new. It can do this for any number of feeds, constantly updating a reading list of your favourite web content.
+
+Importantly, feed readers have no proprietary control over your feed list. If you are dissatisfied with your reader, you can export your feeds to an [OPML](https://opml.org/spec2.opml) (Outline Processor Markup Language) file and import them to a new reader later.
```
Oh I didn’t know that was possible, neat!

ZoeBijl commented on hell/adventcalendar/2025/15/index.md:
```diff
@@ -1,35 +1,99 @@
+  - label: "Mastodon"
+    url: "https://hachyderm.io/@maureenholland"
+    link_label: "@maureenholland"
+intro: "<p>Web feeds are incredible! And a bit confusing! This post offers a peek behind the curtain of what makes a web feed and delights in the fact RSS Autodiscovery means you don’t need to know anything about that to subscribe</p>"
```
add period

```suggestion
intro: "<p>Web feeds are incredible! And a bit confusing! This post offers a peek behind the curtain of what makes a web feed and delights in the fact RSS Autodiscovery means you don’t need to know anything about that to subscribe.</p>"
```

ZoeBijl reviewed (changes requested):
Thank you for your submission, it was fun to read a bit more about RSS feeds.

I would love to learn how you use web feeds. Perhaps this could be a closing section?

Some questions you can use as inspiration:

- What kind of feeds do you follow?
- What feeder do you use?
- Do you check it regularly? Daily even?
- Have you been using feeds for a long time?
- What’s your favourite feed?
- Have you ever come across a feed with easter eggs?

maureenlholland commented:
Thanks for the feedback @ZoeBijl ! Just want to confirm I've seen it and will make updates by Nov. 25

maureenlholland committed:
Fix punctuation

Co-authored-by: Zoë Bijl <5457769+ZoeBijl@users.noreply.github.com>

maureenlholland committed:
Rewrite broadcast syndication paragraph

maureenlholland committed:
Add more on how I use web feeds

maureenlholland commented:
@ZoeBijl ready for re-review when you are

matuzo commented:
Great, thank you!
